Sources/Instrumentation/Crash/KSCrashInstrumentation.swift

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import Foundation
+import OpenTelemetryApi
+
+#if canImport(KSCrashRecording)
+  import KSCrashRecording
+#elseif canImport(KSCrash)
+  import KSCrash
+#endif
+
+#if canImport(KSCrashFilters)
+  import KSCrashFilters
+#endif
+
+import Sessions
+
+protocol CrashProtocol {
+  static func install()
+  static func cacheCrashContext(session: Session?)
+
+  static func recoverCrashContext(from rawCrash: [String: Any],
+                                  log: LogRecordBuilder,
+                                  attributes: [String: AttributeValue]) -> [String: AttributeValue]
+  static func processStoredCrashes()
+}
+
+public class KSCrashInstrumentation: CrashProtocol {
+  public internal(set) static var maxStackTraceBytes = 25 * 1024 // 25 KB
+  public private(set) static var isInstalled: Bool = false
+  private static let logger = OpenTelemetry.instance.loggerProvider.get(instrumentationScopeName: "io.opentelemetry.kscrash")
+  static let reporter = KSCrash.shared
+  static var observers: [NSObjectProtocol] = []
+  private static let queue = DispatchQueue(label: "io.opentelemetry.kscrash", qos: .utility)
+  private static let timestampFormatter: ISO8601DateFormatter = {
+    // Example KSCrash timestamp: `2025-10-28T03:30:53.604204Z`
+    let formatter = ISO8601DateFormatter()
+    formatter.formatOptions = [
+      .withInternetDateTime,
+      .withFractionalSeconds
+    ]
+    return formatter
+  }()
+
+  public init(maxStackTraceBytes: Int = 25 * 1024) {
+    KSCrashInstrumentation.maxStackTraceBytes = maxStackTraceBytes
+    KSCrashInstrumentation.install()
+  }
+
+  public static func install() {
+    guard !isInstalled else {
+      return
+    }
+
+    do {
+      let config = KSCrashConfiguration()
+      config.enableSigTermMonitoring = false
```

Comment:
This is a good default since SIGTERM is not directly actionable in most cases, it's not a programmer error or a "real" issue visible to users.